peps/pep-0795.rst

```diff
@@ -0,0 +1,1245 @@
+PEP: 795
+Title: Deep Immutability in Python
+Author: Matthew Johnson <matjoh@microsoft.com>,
+        Matthew Parkinson <mattpark@microsoft.com>,
+        Sylvan Clebsch <sylvan.clebsch@microsoft.com>,
+        Fridtjof Peer Stoldt <fridtjof.stoldt@it.uu.se>,
+        Tobias Wrigstad <tobias.wrigstad@it.uu.se>
+Sponsor: Michael Droettboom <mike@droettboom.com>
+Discussions-To: https://discuss.python.org/t/96014
+Status: Draft
+Type: Standards Track
+Created: 19-Jun-2025
+Python-Version: 3.15
+Post-History:
+
+
+Abstract
+========
+
+This PEP proposes adding a mechanism for deep immutability to
+Python. The mechanism requires some changes to the core language,
+but user-facing functions are delivered in a module called
+``immutable``. This module provides the following functions and types:
+
+1. The function ``freeze(obj)`` -- turns ``obj`` deeply immutable
+
+2. The function ``isfrozen(obj)`` -- returns ``True`` if ``obj`` is immutable
+
+3. The type ``NotFreezable`` which is an empty type which cannot be made immutable and can be used as a super class to classes whose instances should not be possible to freeze
+
+4. The type ``NotFreezableError`` which is raised on an attempt to mutate an immutable object
```

Comment:
Isn't this for an attempt to freeze and unfreezable object?  Below, it says it raises a `TypeError` when trying to mutate an immutable object.